presto-main/src/main/java/com/facebook/presto/execution/SqlQueryManager.java

```diff
@@ -350,8 +356,26 @@ public static Statement unwrapExecuteStatement(Statement statement, SqlParser sq
             return statement;
         }
 
-        String sql = session.getPreparedStatementFromExecute((Execute) statement);
-        return sqlParser.createStatement(sql);
+        Execute execute = (Execute) statement;
+        String sql = session.getPreparedStatementFromExecute(execute);
+        Statement unwrapped = sqlParser.createStatement(sql);
+
+        // validate that we have the right number of parameters
+        validateParameters(unwrapped, execute.getParameters());
+
+        // replace Parameter expressions with the supplied values.
```

Comment:
Nit: don't end this in a period (it doesn't start with a capital so it isn't a sentence)